File: C/rotate_array.c
void rotate(int* nums, int numsSize, int k){
    if(k > numsSize) k = k % numsSize;
        
    int* tmp = malloc(numsSize * sizeof(int));
        
    for(int i = k; i > 0; i--) {
        *(tmp+(k-i)) = *(nums+(numsSize-i));
    }
        
    for(int i = 0; i < numsSize-k; i++) {
        *(tmp+(k+i)) = *(nums+i);
    }
    
    int a = 0;
    while(a < numsSize) {
        *(nums+a) = *(tmp+a);
        a++;
    }
    
    free(tmp);
}

File: Python3/rotate_array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if k > len(nums): k %= len(nums)
        
        tmp = [0] * len(nums)
        
        for i in range(k, 0 ,-1):
            tmp[k-i] = nums[len(nums)-i]
        
        for i in range(len(nums)-k):
            tmp[k+i] = nums[i]
            
        nums[:] = tmp
        
        del tmp
